src/elements/content-sidebar/BoxAISidebarContent.tsx

```diff
@@ -0,0 +1,150 @@
+    const { questions: cacheQuestions } = cache;
+
+    if (cache.encodedSession !== encodedSession) {
+        setCacheValue('encodedSession', encodedSession);
+    }
+
+    if (cache.questions !== questions) {
+        setCacheValue('questions', questions);
+    }
+
+    React.useEffect(() => {
+        if (!encodedSession && createSession) {
+            createSession();
+        }
+
+        if (cacheQuestions.length > 0 && cacheQuestions[cacheQuestions.length-1].isCompleted === false) {
+            // if we have cache with question that is not completed resend it to trigger an API
+            sendQuestion({prompt: cacheQuestions[cacheQuestions.length-1].prompt});
+        }
+
+        return () => {
+            // stop API request on unmount (e.g. during switching to another tab)
+            stopQuestion();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const renderBoxAISidebarTitle = () => {
+        return (
+            <div className="bcs-BoxAISidebar-title-part">
+                <Text as="h3" className="bcs-title">
+                    {formatMessage(messages.sidebarBoxAITitle)}
+                </Text>
+                {isAIStudioAgentSelectorEnabled &&
+                    <BoxAiAgentSelectorWithApi
+                        fetcher={getAIStudioAgents}
+                        hostAppName={hostAppName}
+                        onSelectAgent={onSelectAgent}
+                        recordAction={recordAction}
+                        shouldHideAgentSelectorOnLoad
+                        // @ts-ignore variant will be available in higher version
+                        variant="sidebar"
+                    />
+                }
+            </div>
+        );
+    };
+
+    const renderActions = () => (
+        <>
+            { renderBoxAISidebarTitle() }
+            { isResetChatEnabled && 
+                <IconButton
+                    aria-label={formatMessage(sidebarMessages.boxAISidebarClear)}
+                    icon={Trash}
+                    onClick={onClearAction}
+                    size="x-small"
+                />
+            }
+        </>
+    );
+
+    return (
+        <AgentsProvider>
+            <SidebarContent 
+                actions={renderActions()}
+                className="bcs-BoxAISidebar"
+                elementId={elementId}
+                sidebarView={SIDEBAR_VIEW_BOXAI}
+            >
+                <div className="bcs-BoxAISidebar-content">
+                    <BoxAiContentAnswers
+                        className="bcs-BoxAISidebar-contentAnswers"
+                        contentName={contentName}
+                        contentType={formatMessage(messages.sidebarBoxAIContent)}
+                        hostAppName={hostAppName}
+                        isAIStudioAgentSelectorEnabled={isAIStudioAgentSelectorEnabled}
+                        isResetChatEnabled={isResetChatEnabled}
```

Comment:
if we are using a custom reset chat callback, do we need to enable the based chat clear button?

Reply:
We don't, good point!